domain/src/main/java/uk/gov/hmcts/ccd/definition/store/domain/displaycontextparameter/DisplayContextParameterType.java

```diff
@@ -0,0 +1,48 @@
+package uk.gov.hmcts.ccd.definition.store.domain.displaycontextparameter;
+
+import com.google.common.base.Strings;
+
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public enum DisplayContextParameterType {
+    DATETIMEENTRY,
+    DATETIMEDISPLAY,
+    TABLE,
+    LIST;
+
+    private static final Pattern PATTERN = Pattern.compile("#(.+)\\((.+)\\)");
+    private static final int TYPE_GROUP = 1;
+    private static final int VALUE_GROUP = 2;
+
+    public static Optional<DisplayContextParameter> getDisplayContextParameterFor(String displayContextParameter) {
```

Comment:
it shouldn't be DisplayContextParameterType's responsibility creating instances of `DisplayContextParameter`. This method should probably be on DisplayContextParameter itself and DisplayContextParameterType should not know about `DisplayContextParameter` but only concern about the type